README.md

```diff
@@ -1,7 +1,205 @@
+1. Production bundles and assets will be written to the `build` directory. This directory is wiped clean each time a production build is initiated.
+1. We use `HtmlWebpackPlugin` to dynamically insert generated bundles to `index.html` during production build. For this reason, you won't find any JavaScript bundle references in `src/index.html`. The resulting production version of `index.html` will be written to the `build` directory. If you inspect that file, you'll see the dynamically generated `<script>` tags with our bundle(s).
+
+#### Bundle analysis
+We have configured `BundleAnalyzerPlugin` to facilitate bundle analysis. This can be useful for managing bundle bloat or spotting unnecessary dependencies. For convenience, a yarn script has been created to launch the bundle analyzer tool. To run this, execute the following command:
+
+```yarn analyze-bundle```
+
+**Note**: you will need to have either started the development server or executed a production build at least one time for this to work as the `build/report.html` file is generated during either of these actions.
+
+### Jest
+[Jest](https://facebook.github.io/jest/) is our testing framework. It provides a friendly testing API, a powerful and easy-to-use mocking functionality, and plenty of speed.
+
+To execute tests, see [Linting & running tests](#linting--running-tests).
+
+To add new tests, create a file with the same name as the file your are testing and append the extension `.test.js`. This file should be located in the same directory as the file your are testing.
+
+For example, if you are testing the `Recidiviz` component which is defined in `Recidiviz.js`, you would add tests in `Recidiviz.test.js`.
+
+See Jest [API](https://facebook.github.io/jest/docs/en/api.html) and [Docs](https://facebook.github.io/jest/docs/en/getting-started.html) for more information.
+
+### eslint
+[eslint](https://eslint.org/) is a flexible linter for JavaScript. We have configured eslint to adhere to the Airbnb style guides for [Javascript](https://github.com/airbnb/javascript) and [React](https://github.com/airbnb/javascript/tree/master/react), with a few [exceptions](https://github.com/Recidiviz/recidiviz.com/.eslintrc), in addition to the base eslint JS rules.
+
+To run the linter, see [Linting & running tests](#linting--running-tests).
+
+We suggest installing a linting package for your preferred code editor that hooks into eslint. This will allow you to get real time feedback on code while you're writing it. We recommend [linter-eslint](https://atom.io/packages/linter-eslint) for Atom.
+
+### husky
+[husky](https://github.com/typicode/husky) is a package that provides easy-to-use Git hooks. These are defined in `package.json` under the `husky` property. We have configured husky to run linting and testing prior to all git commits and pushes using the `pre-commit` and `pre-push` hooks.
```

Comment:
So over on https://github.com/Recidiviz/recidiviz-data/issues/53, I filed an issue to consider pre-commit (or pre-push) git hooks across all Recidiviz projects. After getting continuous integration with linting and testing up and running on recidiviz-data, my thought was to wait and see whether or not I thought that hooks which ran those steps would be useful or an impedance, especially given that it will be done automatically on each new push to each branch and with each PR, and that we control merge access. My concern was, generally, that if there is an unforeseen issue causing a lint or test failure on a local machine that was phantasmic or that we otherwise were unconcerned with, being unable to commit would be unnecessary friction. 

But what are your thoughts here? I'm also happy to do a bake-off where we invoke the commits here and not on recidiviz-data, and then gauge how contributors are feeling later.

Reply:
I take your point about blocking progress.

I personally like enforcing these to catch issues as far upstream as possible. One nice thing is that you can bypass the checks, if you really want to, using the `--no-verify` flag. That could let us get the benefit of local verification while preventing us from being blocked in a crunch.

Down for the bake-off for sure.

Reply:
Ah, right, I forgot `--no-verify`. Bake off!